src/Interface.jl

```diff
@@ -0,0 +1,46 @@
+
+
+## Some function which make the model easier to define.
+
+
+# A very light weight container for a state space model
+# The state space is one dimensional!
+# Note that this is only for a very simple demonstration.
+mutable struct Container
+    x::Vector{Float64}
+    num_produce::Float64
+end
+
+# This is important for initalizaiton
+function init()
```

Comment:
I guess you could just write `const init = current_trace`?

Reply:
I am sorry, i did not mention that the interface is only for debugging and demonstration. Maybe i should remove it completely out of this version.

Reply:
No worries, just ignore all my comments regarding the debugging functionality :slightly_smiling_face: